lib/awful/spawn.lua

```diff
@@ -79,6 +79,52 @@
 --        placement = awful.placement.bottom_right,
 --    })'
 --
+-- **Getting a command result**:
+--
+-- First, do **not** use `io.popen` **ever**. It is synchronous. Synchronous
+-- function **stop everything** until they are done. All visual applications
+-- locks (as Awesome no longer respond), you will probably lose some keyboard
+-- and mouse events and will have higher latency when playing games. This is
+-- also true about reading files, but this is another topic.
+--
+-- Awesome provides a few ways of getting command results. One is to use the
+-- `Gio` libraries directly. This is usually very complicated, but gives a lot
+-- on control over the command execution, including the ability to create
+-- custom pipes and link many process.
```

Comment:
Uhm, I'm not sure if `Gio` really provides this, and if it does, then I bet that this is not accessible through `lgi`. Could we remove this part? (And just say that using `Gio` directly is complicated)